### PR TITLE
Adds tests for the extensions

### PR DESCRIPTION
**Pull request by pms1969**
# Pull Request Checklist

- [x] Read the [contributing guidelines](../blob/master/docs/CONTRIBUTING.md)
- [x] Tests pass

# Description of the PR

Adds tests that can be run against the SDK extensions without too much of a dependency on any single SDK.

### CODE REVIEW

File: tests/Tests/Integration/TokenProviderConfigurationTest.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    {
+        private static readonly Lazy<ApiConfiguration> ApiConfig =
+            new Lazy<ApiConfiguration>(() => ApiConfigurationBuilder.Build("secrets.json"));
+
+        //Test requires [assembly: InternalsVisibleTo("namespace Finbourne.Sdk.Extensions.IntegrationTests")] in SDK project.
+        [Test]
+        public void Construct_AccessToken_NonNull()
+        {
+            ITokenProvider tokenProvider;
+            if (ApiConfig.Value.MissingSecretVariables)
```
Comment (rizwansaeed) [inline tests/Tests/Integration/TokenProviderConfigurationTest.cs:17]: should this be 2 tests?

where/how is the config set? i dont see it being set in the test/test setup

Comment (pms1969) [reply tests/Tests/Integration/TokenProviderConfigurationTest.cs:17]: good question, was just copy-pasta. so there's some chance it's relying on state for the integration-y ones.

**Comment by adamMQM** [comment]
gonna switch branches and back due to https://github.com/isaacs/github/issues/750

File: tests/Tests/Integration/TokenProviderConfigurationTest.cs (hunk 1 above)
Comment (adamMQM) [reply tests/Tests/Integration/TokenProviderConfigurationTest.cs]: I've split this out into two separate tests now

**Review by rizwansaeed — APPROVED** [review]

**Pull request merged**